In [1]:
import pandas as pd

In [2]:
#reading in csv files
apple = pd.read_csv('./datasets/dataset_with_kaggle_pics/apple.csv')
lemon = pd.read_csv('./datasets/dataset_with_kaggle_pics/lemon.csv')
bellpepper = pd.read_csv('./datasets/dataset_with_kaggle_pics/bellpepper.csv')
onion = pd.read_csv('./datasets/dataset_with_kaggle_pics/onion.csv')
pear = pd.read_csv('./datasets/dataset_with_kaggle_pics/pear.csv')
strawberry = pd.read_csv('./datasets/dataset_with_kaggle_pics/strawberry.csv')
tomato = pd.read_csv('./datasets/dataset_with_kaggle_pics/tomato.csv')

In [3]:
#sanity check
print (f'apple null count {apple.isnull().sum().sum()}')
print (f'lemon null count {lemon.isnull().sum().sum()}')
print (f'bellpepper null count {bellpepper.isnull().sum().sum()}')
print (f'onion null count {onion.isnull().sum().sum()}')
print (f'pear null count {pear.isnull().sum().sum()}')
print (f'strawberry null count {strawberry.isnull().sum().sum()}')
print (f'tomato null count {tomato.isnull().sum().sum()}')

apple null count 0
lemon null count 0
bellpepper null count 0
onion null count 0
pear null count 0
strawberry null count 0
tomato null count 0


In [4]:
#combine all the dataframes together
frames = [apple, lemon, bellpepper, onion, pear, strawberry, tomato]

df = pd.concat(frames)
df.head()

pixel0  pixel1  pixel2  pixel3  pixel4  pixel5  pixel6  pixel7  pixel8  \
0       0       0       0       0       0       0       0       0       0   
1      38      39      73      35      51      93      40      84     138   
2       5       4       6       5       4       6       5       4       6   
3       6      82     232       1      74     218       0      62     200   
4     144     175     200     136     167     200     105     131     185   

   pixel9  ...  pixel29991  pixel29992  pixel29993  pixel29994  pixel29995  \
0       0  ...           1           0           2           1           0   
1      44  ...          19          30         142          25          37   
2       6  ...           1           1           1           1           1   
3       0  ...           0           0          10           0           0   
4      77  ...          72          68         223          65          61   

   pixel29996  pixel29997  pixel29998  pixel29999  target  
0           2           1           0           2   apple  
1         149          30          42         154   apple  
2           1           1           1           1   apple  
3          11           0           0          12   apple  
4         217          52          48         204   apple  

[5 rows x 30001 columns]

In [5]:
#extra sanity check
df.isnull().sum().sum()

0

#### I should have 24,881 rows and 30,001 columns

In [6]:
# sanity check
df.shape

(24881, 30001)

#### I should have 7 distinct target categories

In [7]:
#sanity check 
len(df['target'].unique())

7

In [9]:
#extra sanity check
df.isnull().sum().sum()

0

# Time to Shuffle the Data

In [8]:
from sklearn.utils import shuffle

In [9]:
#before shuffling 
df.head()

pixel0  pixel1  pixel2  pixel3  pixel4  pixel5  pixel6  pixel7  pixel8  \
0       0       0       0       0       0       0       0       0       0   
1      38      39      73      35      51      93      40      84     138   
2       5       4       6       5       4       6       5       4       6   
3       6      82     232       1      74     218       0      62     200   
4     144     175     200     136     167     200     105     131     185   

   pixel9  ...  pixel29991  pixel29992  pixel29993  pixel29994  pixel29995  \
0       0  ...           1           0           2           1           0   
1      44  ...          19          30         142          25          37   
2       6  ...           1           1           1           1           1   
3       0  ...           0           0          10           0           0   
4      77  ...          72          68         223          65          61   

   pixel29996  pixel29997  pixel29998  pixel29999  target  
0           2           1           0           2   apple  
1         149          30          42         154   apple  
2           1           1           1           1   apple  
3          11           0           0          12   apple  
4         217          52          48         204   apple  

[5 rows x 30001 columns]

In [10]:
#after shuffling
df = shuffle(df)
df.head()

pixel0  pixel1  pixel2  pixel3  pixel4  pixel5  pixel6  pixel7  pixel8  \
1340     254     255     251     254     255     251     254     255     251   
3584     255     255     255     255     255     255     255     255     255   
1463     255     255     255     255     255     255     255     255     255   
4615     255     249     255     255     250     255     254     251     255   
283        8      14      13       9      15      14      10      15      14   

      pixel9  ...  pixel29991  pixel29992  pixel29993  pixel29994  pixel29995  \
1340     254  ...         255         255         255         255         255   
3584     255  ...         255         255         255         255         255   
1463     255  ...         255         255         255         255         255   
4615     254  ...         255         255         255         255         255   
283       10  ...          30          22          23          30          22   

      pixel29996  pixel29997  pixel29998  pixel29999      target  
1340         255         255         255         255  bellpepper  
3584         255         255         255         255        pear  
1463         255         255         255         255        pear  
4615         255         255         255         255       apple  
283           23          31          23          24       lemon  

[5 rows x 30001 columns]

In [11]:
#reseting index 
df.reset_index(drop = True, inplace = True)
df.head()

pixel0  pixel1  pixel2  pixel3  pixel4  pixel5  pixel6  pixel7  pixel8  \
0     254     255     251     254     255     251     254     255     251   
1     255     255     255     255     255     255     255     255     255   
2     255     255     255     255     255     255     255     255     255   
3     255     249     255     255     250     255     254     251     255   
4       8      14      13       9      15      14      10      15      14   

   pixel9  ...  pixel29991  pixel29992  pixel29993  pixel29994  pixel29995  \
0     254  ...         255         255         255         255         255   
1     255  ...         255         255         255         255         255   
2     255  ...         255         255         255         255         255   
3     254  ...         255         255         255         255         255   
4      10  ...          30          22          23          30          22   

   pixel29996  pixel29997  pixel29998  pixel29999      target  
0         255         255         255         255  bellpepper  
1         255         255         255         255        pear  
2         255         255         255         255        pear  
3         255         255         255         255       apple  
4          23          31          23          24       lemon  

[5 rows x 30001 columns]

In [12]:
#another sanity check
df.shape

(24881, 30001)

In [13]:
df.groupby('target')['target'].count()
#another sanity check
#this should match the number of images per produce item

target
apple         7806
bellpepper    2571
lemon         1988
onion         1987
pear          4485
strawberry    2555
tomato        3489
Name: target, dtype: int64

**Same table from README**

|Produce Name|Image QTY|
|--|--|
|Apple|7,806|
|Bellpepper|2,571|
|Lemon|1,988|
|Onion|1,987|
|Pear|4,485|
|Strawberry|2,555|
|Tomato|3,489|
|**Total**|**24,881**|

In [14]:
#another sanity check
df.head()

pixel0  pixel1  pixel2  pixel3  pixel4  pixel5  pixel6  pixel7  pixel8  \
0     254     255     251     254     255     251     254     255     251   
1     255     255     255     255     255     255     255     255     255   
2     255     255     255     255     255     255     255     255     255   
3     255     249     255     255     250     255     254     251     255   
4       8      14      13       9      15      14      10      15      14   

   pixel9  ...  pixel29991  pixel29992  pixel29993  pixel29994  pixel29995  \
0     254  ...         255         255         255         255         255   
1     255  ...         255         255         255         255         255   
2     255  ...         255         255         255         255         255   
3     254  ...         255         255         255         255         255   
4      10  ...          30          22          23          30          22   

   pixel29996  pixel29997  pixel29998  pixel29999      target  
0         255         255         255         255  bellpepper  
1         255         255         255         255        pear  
2         255         255         255         255        pear  
3         255         255         255         255       apple  
4          23          31          23          24       lemon  

[5 rows x 30001 columns]

## Let's Save the Data!

In [19]:
df.to_csv('./datasets/mama.csv', index = False)